src/main/java/org/opensearch/security/support/HTTPHelper.java

```diff
@@ -38,11 +38,17 @@
 
 public class HTTPHelper {
 
-    public static AuthCredentials extractCredentials(String authorizationHeader, Logger log) {
+    public static AuthCredentials extractCredentials(String authorizationHeader, Logger log, boolean isChallenge) {
+
+        boolean isTraceEnabled = log.isTraceEnabled();
 
         if (authorizationHeader != null) {
             if (!authorizationHeader.trim().toLowerCase().startsWith("basic ")) {
-                log.warn("No 'Basic Authorization' header, send 401 and 'WWW-Authenticate Basic'");
+                if (isChallenge) {
```

Comment:
Apologies for joining the party late, i'm slightly confused on what exactly are we trying to solve here? Is it that we log at warn only when Basic Auth fails or is  it that we will log at warn when no other domains to be evaluated against are present?

Reply:
If the idea is to limit logging of this message only when basic auth is the challenging authenticator and creds fail, I'm okay with this flag.